system/lib/libc/musl/src/env/__environ.c

```diff
@@ -1,6 +1,42 @@
 #include "libc.h"
 
+#ifdef __EMSCRIPTEN__
+#include <stdlib.h>
+#include <wasi/wasi.h>
+#endif
+
 char **__environ = 0;
 weak_alias(__environ, ___environ);
 weak_alias(__environ, _environ);
 weak_alias(__environ, environ);
+
+#ifdef __EMSCRIPTEN__
+__attribute__((constructor(100))) // construct this before user code
+void __emscripten_environ_constructor(void) {
```

Comment:
Eventually I'd like to see this happens in crt1.c as part of _start (which could weakly depend on this function so it could still be DCEd). 

Calling static ctors can eventually happen from there too.